src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.ShoppingCartDao;
+import com.cinema.exceptions.DataProcessingException;
+import com.cinema.lib.Dao;
+import com.cinema.model.ShoppingCart;
+import com.cinema.model.User;
+import com.cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Failed to add the shopping cart "
+                    + shoppingCart + " to the DB.", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery("from ShoppingCart sc "
+                    + "join fetch sc.user u "
+                    + "left join fetch sc.tickets t "
+                    + "where u.id = :userId", ShoppingCart.class);
+            query.setParameter("userId", user.getId());
```

Comment:
You can just use the user and check user objects. Hibernate will take ids anyway

Reply:
Thanks))